### PR TITLE
JSRS migration

### PR DESCRIPTION
**Pull request by gabrieloliveira41**

### CODE REVIEW

**Comment by gabrieloliveira41** [comment]
Goals verified after changes (7 out of 7 goals total):
- CommitOrder
Buy 7 polo shirts and 10 henley tops
Buy 7 XL t-shirts, 6 XS henley tops, and 5 XS pullover sweatshirts
Order a pullover
- CancelCommittedOrder
Cancel my order
- Shirt
The polo
Oxford dress shirt
- Order#totalPrice
What's the total
What is the total price of the order?
- CancelOrder
Cancel this order
- Size
I'm an extra large
- CheckStatus
Check on my shirt order

All tests resulted in equal intents and equal IDE simulator results after migration comparing to before the migration

**Comment by gabrieloliveira41** [comment]
after the migration there were some files that there were using `import dates` so i had to add the lib `zoned-date-time-polyfill.js` and update the files `CancelCommittedOrder.js` `CheckStatus.js` `CommitOrder.js` `CreateOrder.js` for the new lib format

**Comment by gabrieloliveira41** [comment]
on file `CreateAddedItems.js` had the following error 
image
and that same error appeared on files `CreateChangedItems.js` `CreateItems.js` `CreateRemovedItems.js` so i used the same solution which as declaring the variable items 
image

**Comment by gabrieloliveira41** [comment]
on file `UpdateOrder.js` on line 46 there were the following error 
image
so i declared the variable `oldQuantity` and then there no errors left reported by the IDE

**Comment by gabrieloliveira41** [comment]
the only diference i noticed on the capsule compared to V1 was that the time was a diferent, there were a 10 minutes diference on the time, so i reported that to thiago amaral (sidi) and he made a quick fix on the `plusSeconds `function and the `plusHours` function, after that the capsule was running as expected

**Comment by gabrieloliveira41** [comment]
@kangawoo44 @abapayeViv could you review my PR, please?

**Comment by kangawoo44** [comment]
> the only diference i noticed on the capsule compared to V1 was that the time was a diferent, there were a 10 minutes diference on the time, so i reported that to thiago amaral (sidi) and he made a quick fix on the `plusSeconds `function and the `plusHours` function, after that the capsule was running as expected

have you tried using the latest library version 1.1.4? is it an issue there as well?

File: resources/base/endpoints.bxb [hunk 1]
```diff
@@ -7,7 +7,7 @@ endpoints {
       local-endpoint ("FindShirt.js")
     }
     action-endpoint (CreateOrder) {
-      accepted-inputs (initialItems)
+      accepted-inputs (initialItems,$vivContext)
```
Comment (kangawoo44) [inline resources/base/endpoints.bxb]: add space between the params. apply to all changes on this file.

**Comment by gabrieloliveira41** [comment]
> > the only diference i noticed on the capsule compared to V1 was that the time was a diferent, there were a 10 minutes diference on the time, so i reported that to thiago amaral (sidi) and he made a quick fix on the `plusSeconds `function and the `plusHours` function, after that the capsule was running as expected
> 
> have you tried using the latest library version 1.1.4? is it an issue there as well?

i tried but there is 8 errors reported by the IDE on that version so i decided to use the version 1.1.3 [sidi]
verison 1.1.4 errors:
image

**Comment by kangawoo44** [comment]
can we find out what lexical declaration it's throwing the error on? This is something we should report back to the polyfill library owner as a bug. And if we can resolve those errors I'd like to also see if there's a time difference there or not.

Update:
I’ve looked into it and it seems like the issue is caused by lexical scoping of switch https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#lexical_scoping
and it seems like a simple {} wrap around the default case statements will do the trick.

**Comment by gabrieloliveira41** [comment]
after fixing Dates Library v1.1.4 by adding the {} and the capsule worked as expected and showed the correct time as v1.1.3 [sidi]
@kangawoo44 do you prefer to use which version for the capsule?

**Comment by kangawoo44** [comment]
Please use the official version in all migrations so we can continue to find bugs and fix them. We should only use our own versions temporarily before issues haven't been fixed yet.

**Review by kangawoo44 — APPROVED** [review]

**Review by abapayeViv — APPROVED** [review]
👍

**Pull request merged**